proto/process/agent.proto

```diff
@@ -570,35 +578,109 @@ message ServiceSessionAffinityConfig {
     int32 clientIPTimeoutSeconds = 1;
 }
 
+// reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/core/v1/generated.proto#L2107
+message Node {
+	Metadata metadata = 1;
+	NodeSpec spec = 2;
+	NodeStatus status = 3;
+	bytes yaml = 4;
+	repeated string tags = 5;
+	Host host = 6; // Used post-resolution
+}
+
+// reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/core/v1/generated.proto#L2312
+message NodeSpec {
+	string podCIDR = 1;
+	repeated string podCIDRs = 3;
+	string providerID = 2;
+	bool unschedulable = 4;
+	repeated Taint taints = 5;
+}
+
+// reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/core/v1/generated.proto#L2348
+// skipping NodeConfigStatus for now.
+message NodeStatus {
+	// Key/Value of a resource type and a corresponding quantity as an int64.
+	map<string, int64> capacity = 1;
+	// Key/Value of a resource type and a corresponding quantity as an int64.
+	map<string, int64> allocatable = 2;
+	// https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/core/v1/generated.proto#L2127
+	map<string,string> nodeAddresses = 3;
+	repeated NodeCondition conditions = 4;
```

Comment:
Could you check how the `status` column is generated for nodes in `kubectl get`? I think we'll want to treat that somewhat like pods, for which the status is the `reason` field of the first condition that failed, and we also store the message about it so we'll be able to add it as a tooltip for quick access, but then we discard the rest. See:
- https://github.com/DataDog/agent-payload/blob/5511817bc817e6031ee5426c3c8771ed56adca38/proto/process/agent.proto#L738-L739
- https://github.com/DataDog/datadog-agent/blob/d5f940ef3ba93fec618705577858954283265a93/pkg/process/util/orchestrator/pod.go#L256-L324
- and PodCondition that we dropped: https://github.com/kubernetes/kubernetes/blob/release-1.19/staging/src/k8s.io/api/core/v1/generated.proto#L3079-L3104

Reply:
This seems doable. But we do lose some metadata. Which for semi/non volatile things like nodes could be possibly interesting. 
lost metadata: https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/core/v1/generated.proto#L2174-L2180

e.g. upstream print node:
https://github.com/kubernetes/kubernetes/blob/1e12d92a5179dbfeb455c79dbf9120c8536e5f9c/pkg/printers/internalversion/printers.go#L1410

Reply:
Added both (NodeConditions slice) + self computed status as in `k get nodes`
E.g. output
```
❯ k get nodes
NAME                                      STATUS   ROLES    AGE   VERSION
gke-nam-test-default-pool-d718e7e7-24lp   Ready    <none>   20d   v1.16.13-gke.1
```